src/main/java/org/mule/extension/mulechain/helpers/AwsbedrockEmbeddingPayloadHelper.java

```diff
@@ -187,12 +196,26 @@ private static BedrockRuntimeClient createClient(AwsbedrockConfiguration configu
                     configuration.getAwsSecretAccessKey(),
                     configuration.getAwsSessionToken());
         }
-
-        return BedrockRuntimeClient.builder()
+        
+        String endpointOverride = configuration.getEndpointOverride();
+        
+        SdkHttpClient httpClient = CommonUtils.buildHttpClientWithTimeout(
+        	    configuration.getTimeout(),
+        	    configuration.getTimeoutUnit()
+        	);
+
+        
+		BedrockRuntimeClientBuilder builder = BedrockRuntimeClient.builder()
                 .credentialsProvider(StaticCredentialsProvider.create(awsCredentials))
                 .region(region)
-                .fipsEnabled(configuration.getFipsModeEnabled())
-                .build();
+                .httpClient(httpClient)
+                .fipsEnabled(configuration.getFipsModeEnabled());
+
+        if (endpointOverride != null && !endpointOverride.isEmpty()) {
```

Comment:
Generally speaking, it's better to check for `isBlank()` instead of `isEmpty()` ... the latter will cause failures if they accidentally enter a space for `endpointOverride`.